app/src/main/java/com/github/fribourgsdp/radio/UserProfileActivity.kt

```diff
@@ -101,15 +162,19 @@ class UserProfileActivity : AppCompatActivity(), PlaylistAdapter.OnPlaylistClick
     private fun checkUser(){
         //get current user
         val firebaseUser = firebaseAuth.currentUser
-        if(firebaseUser == null){
-            //user not logged in
-            startActivity(Intent(this, GoogleSignInActivity::class.java))
-            finish()
-        }else{
-            //user logged in
-            //get user info
-            /* TODO MAP FIREBASE INFO TO USER*/
-
+        if(firebaseUser != null){
```

Comment:
It makes more sense to do this in google sign in where you can retrieve this information directly at login.

Reply:
I don't fully understand what would be the main difference. Could you give a small explanation @BartLucien?

Reply:
@BartLucien explained this to me in real life and I agree.